src/generala/utils/CombinationFinder.java

```diff
@@ -0,0 +1,240 @@
+        }
+        combinationTreeMap.put(CombinationEnum.PAIR, (biggestPair * PAIR_SIZE) + CombinationEnum.PAIR.getScoreConst());
+    }
+
+    private void addDoublePair(EnumSet<CombinationEnum> playerRolledCombinations,
+                               Map<CombinationEnum, Integer> combinationTreeMap) {
+        if (hasPlayerRolledCombination(playerRolledCombinations, CombinationEnum.DOUBLE_PAIR)) {
+            return;
+        }
+        combinationTreeMap.put(CombinationEnum.DOUBLE_PAIR,
+                ((biggestPair + secondBiggestPair) * PAIR_SIZE)
+                        + CombinationEnum.DOUBLE_PAIR.getScoreConst());
+
+    }
+
+    private void addFullHouse(EnumSet<CombinationEnum> playerRolledCombinations,
+                              Map<CombinationEnum, Integer> combinationTreeMap) {
+        if (hasPlayerRolledCombination(playerRolledCombinations, CombinationEnum.FULL_HOUSE)) {
+            return;
+        }
+        int pairToUseInCalculatingFullHouse;
+        if (biggestPair != biggestTriple) {
+            pairToUseInCalculatingFullHouse = biggestPair;
+        } else if (hasSecondBiggestPair()) {
+            pairToUseInCalculatingFullHouse = secondBiggestPair;
+        } else {
+            return;
+        }
+        combinationTreeMap.put(CombinationEnum.FULL_HOUSE,
+                (pairToUseInCalculatingFullHouse * PAIR_SIZE) + (biggestTriple * TRIPLE_SIZE)
+                        + CombinationEnum.FULL_HOUSE.getScoreConst());
+    }
+
+    private void addStraight(EnumSet<CombinationEnum> playerRolledCombinations,
+                             Map<CombinationEnum, Integer> combinationTreeMap) {
+        if (hasPlayerRolledCombination(playerRolledCombinations, CombinationEnum.STRAIGHT)) {
+            return;
+        }
+        combinationTreeMap.put(CombinationEnum.STRAIGHT, countStraightScore(sideForCalculatingStraightPoints));
+    }
+
+    //BOOLEAN METHODS
+    private boolean hasGenerala(int currentDieSideDuplicates) {
+        return currentDieSideDuplicates == DiceRoll.getDiceCount();
+    }
+
+    private boolean hasBiggestPair() {
+        return biggestPair != 0;
+    }
+
+    private boolean hasSecondBiggestPair() {
+        return secondBiggestPair != 0;
+    }
+
+    private boolean hasBiggestTriple() {
+        return biggestTriple != 0;
+    }
+
+    private boolean hasBiggestFourOfAKind() {
+        return biggestFourOfAKind != 0;
+    }
+
+    private boolean hasPlayerRolledCombination(EnumSet<CombinationEnum> rolledCombinations,
+                                               CombinationEnum combinationEnum) {
+        return rolledCombinations.contains(combinationEnum);
+    }
+
+    private boolean hasStraight(Set<Integer> rolledSidesSet) {
+        List<Integer> rolledSidesList = new ArrayList<>(rolledSidesSet);
+        for (int i = 0, counter = 0; i < rolledSidesList.size() - 1; i++) {
+            if (rolledSidesList.get(i) - 1 != rolledSidesList.get(i + 1)) {
+                counter = 0;
```

Comment:
Why do you need the counter? Obviously the method does more than one thing - finding if the combination is straight and incrementing counter. Try to separate it if you rely on the counter variable (apply this only if it is possible)

Can we use it without the counter:

`public boolean hasStraight(Set<Integer> rolledSidesSet) {
       List<Integer> rolledNumbers = new ArrayList<>(rolledSidesSet);
        Collections.sort(rolledNumbers);

        for (int i = 0; i < rolledNumbers.size() - 1; i++) {
            if (rolledNumbers.get(i) + 1 != rolledNumbers.get(i + 1)) {
                return false;
            }
        }

        return true;
    }`